server/routes/newspaperArchive.ts

```diff
@@ -36,7 +39,14 @@ router.get('/auth', async (req: Request, res: Response) => {
 	const authString = config?.authString;
 	if (authString === undefined) {
 		log.error(`Missing newspaper archive auth key`);
-		res.status(500).send();
+		return res.sendStatus(500);
+	}
+
+	const hasCorrectEntitlement = await checkSupporterEntitlement(req);
```

Comment:
I wonder if this should be wrapped in a try/catch in case any of the underlying operations error (for example if the request to mdapi fails)? I guess in that case we'd want to set `hasCorrectEntitlement` to false to be on the safe side.

Reply:
I've wrapped the whole thing in a try catch, seems more accurate than eg telling a user they don't have a product in the case the mdapi call failed